backends/aerospike.go

```diff
@@ -13,59 +15,111 @@ import (
 const setName = "uuid"
 const binValue = "value"
 
-type Aerospike struct {
+type AerospikeDB interface {
+	NewUuidKey(namespace string, key string) (*as.Key, error)
+	Get(key *as.Key) (*as.Record, error)
+	Put(key *as.Key, value string, ttlSeconds int) error
+}
+
+type AerospikeDBClient struct {
+	client *as.Client
+}
+
+func (db AerospikeDBClient) Get(key *as.Key) (*as.Record, error) {
+	rec, err := db.client.Get(nil, key, binValue)
+	if err != nil {
+		return nil, formatAerospikeError(err, "GET")
+	}
+	return rec, nil
+}
+
+func (db AerospikeDBClient) Put(key *as.Key, value string, ttlSeconds int) error {
+	bins := as.BinMap{binValue: value}
+	policy := &as.WritePolicy{Expiration: uint32(ttlSeconds)}
+
+	err := db.client.Put(policy, key, bins)
+
+	return formatAerospikeError(err, "PUT")
+}
+
+func (db *AerospikeDBClient) NewUuidKey(namespace string, key string) (*as.Key, error) {
+	asKey, err := as.NewKey(namespace, setName, key)
+	if err != nil {
+		return nil, err
+	}
+	return asKey, nil
+}
+
+type AerospikeBackend struct {
 	cfg     config.Aerospike
-	client  *as.Client
+	client  AerospikeDB
 	metrics *metrics.Metrics
 }
 
-func NewAerospikeBackend(cfg config.Aerospike, metrics *metrics.Metrics) *Aerospike {
+func NewAerospikeBackend(cfg config.Aerospike, metrics *metrics.Metrics) *AerospikeBackend {
+	if cfg.Host == "" {
+		log.Fatalf("Cannot connect to empty Aerospike host")
+	}
+	if cfg.Port <= 0 {
+		log.Fatalf("Cannot connect to Aerospike host at port %d", cfg.Port)
+	}
 	client, err := as.NewClient(cfg.Host, cfg.Port)
 	if err != nil {
-		log.Fatalf("Error creating Aerospike backend: %v", err)
-		panic("Aerospike failure. This shouldn't happen.")
+		log.Fatalf("Error creating Aerospike backend: %v", formatAerospikeError(err, "NewAerospikeBackend"))
```

Comment:
Do we need the caller for this `NewAerospikeBackend` be present in the error message? I think the error message itself seems descriptive enough.

Reply:
Removed